modules/markets-api/src/main/scala/org/ergoplatform/dex/markets/api/v1/services/AmmStats.scala

```diff
@@ -141,9 +142,11 @@ object AmmStats {
                     } yield Fees(feesX.value + feesY.value, UsdUnits, window)
                   case None => OptionT.pure[F](Fees.empty(UsdUnits, window))
                 }
-
         yearlyFeesPercent <- OptionT.liftF(ammMath.feePercentProjection(tvl, fees, info, MillisInYear))
-      } yield PoolSummary(poolId, pool.lockedX, pool.lockedY, tvl, volume, fees, yearlyFeesPercent)).value
+        res <- if (validTokens.contains(pool.lockedX.id) && validTokens.contains(pool.lockedY.id))
```

Comment:
First, it's not necessary to calculate volume, fee etc. if tokens are not in our token list. Next, we should return `PoolSummary` with _empty_ volume, fees etc if tokens are not in list.